gdal/.gitignore

```diff
@@ -5,6 +5,7 @@ detectsse.dSYM
 gdal.pc
 config.log
 config.status
+configure
```

Comment:
Sorry for bothering you again.... I've added `gdal/configure` in `.gitignore` in the gdal root directory... may I ask why this is needed here? Thanks

Reply:
ah I missed that. this is indeed redundant. As the gdal/.gitignore is more comprehensive, I'm going to move the additions from toplevel .gitignore into gdal/.gitignore